Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-138.ca-central-1.compute.internal
Sha: 8286e6382e3aad5a0d8b84403a03409200a6f520
Number of Error(s): 8
/home/centos/smoketest/PR-14705/HPCC-Platform/roxie/udplib/udptrs.cpp:710:214: error: no matching function for call to \xe2\x80\x98UdpReceiverEntry::UdpReceiverEntry(const IpAddress, const IpAddress&, const int&, const int&, const int&, const int&)\xe2\x80\x99
/home/centos/smoketest/PR-14705/HPCC-Platform/roxie/udplib/udptrs.cpp:710:217: error: no matching function for call to \xe2\x80\x98IpMapOf<UdpReceiverEntry>::IpMapOf(CSendManager::CSendManager(int, int, int, int, const IpAddress&, int, int, const IpAddress&, TokenBucket*, bool)::<lambda(const ServerIdentifier&)>)\xe2\x80\x99
make[2]: *** [roxie/udplib/CMakeFiles/udplib.dir/build.make:115: roxie/udplib/CMakeFiles/udplib.dir/udptrs.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:17021: roxie/udplib/CMakeFiles/udplib.dir/all] Error 2
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-03-18_16-14-55
time logs started
Prep time   :     8 sec (00:00:08)
Build time  :   381 sec (00:06:21)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   389 sec (00:06:29)
time logs end

[Build error:](https://gist.github.com/13cd39398a66581bf68f84445ec47cd3)
Build log [Build log](https://gist.github.com/HPCCSmoketest/13cd39398a66581bf68f84445ec47cd3#file-result-21-03-18-16-08-13-log)

--- a/esp/src/package.json
+++ b/esp/src/package.json
@@ -36,6 +36,7 @@
     "@fluentui/react": "^8.1.4",
     "@fluentui/react-cards": "^1.0.0-beta.0",
     "@fluentui/react-hooks": "^8.0.2",
+    "@fluentui/react-icons-mdl2": "^1.0.2",
     "@hpcc-js/chart": "^2.57.0",
     "@hpcc-js/codemirror": "^2.38.0",
     "@hpcc-js/common": "^2.47.0",

--- a/esp/src/src-react/components/WorkunitsDashboard.tsx
+++ b/esp/src/src-react/components/WorkunitsDashboard.tsx
@@ -6,7 +6,7 @@ import * as ESPWorkunit from "src/ESPWorkunit";
 import { WorkunitsService, WUQuery } from "@hpcc-js/comms";
 import { Area, Column, Pie, Bar } from "@hpcc-js/chart";
 import { chain, filter, group, map, sort } from "@hpcc-js/dataflow";
-import Chip from "@material-ui/core/Chip";
+import { Chip } from "./controls/Chip";
 import nlsHPCC from "src/nlsHPCC";
 import { Memory } from "src/Memory";
 import { pushParamExact } from "../util/history";
@@ -206,7 +206,7 @@ export const WorkunitsDashboard: React.FunctionComponent<WorkunitsDashboardProps
                                 <Card.Item>
                                     <Stack horizontal horizontalAlign="space-between">
                                         <Text variant="large" nowrap block styles={{ root: { fontWeight: "bold" } }}>{nlsHPCC.State}</Text>
-                                        {filterProps.state !== undefined && <Chip label={filterProps.state} clickable color="primary" onDelete={() => pushParamExact("state", undefined)} />}
+                                        {filterProps.state !== undefined && <Chip label={filterProps.state} onDelete={() => pushParamExact("state", undefined)} />}
                                     </Stack>
                                 </Card.Item>
                                 <Card.Item>
@@ -219,7 +219,7 @@ export const WorkunitsDashboard: React.FunctionComponent<WorkunitsDashboardProps
                                 <Card.Item>
                                     <Stack horizontal horizontalAlign="space-between">
                                         <Text variant="large" nowrap block styles={{ root: { fontWeight: "bold" } }}>{nlsHPCC.Day}</Text>
-                                        {filterProps.day !== undefined && <Chip label={filterProps.day} clickable color="primary" onDelete={() => pushParamExact("day", undefined)} />}
+                                        {filterProps.day !== undefined && <Chip label={filterProps.day} color="primary" onDelete={() => pushParamExact("day", undefined)} />}
                                         <Dropdown onChange={(evt, opt, idx) => { pushParamExact("lastNDays", opt.key); }}
                                             options={[
                                                 { key: 1, text: "1 Day", selected: filterProps.lastNDays === 1 },
@@ -243,7 +243,7 @@ export const WorkunitsDashboard: React.FunctionComponent<WorkunitsDashboardProps
                                 <Card.Item>
                                     <Stack horizontal horizontalAlign="space-between">
                                         <Text variant="large" nowrap block styles={{ root: { fontWeight: "bold" } }}>{nlsHPCC.Protected}</Text>
-                                        {filterProps.protected !== undefined && <Chip label={"" + filterProps.protected} clickable color="primary" onDelete={() => pushParamExact("protected", undefined)} />}
+                                        {filterProps.protected !== undefined && <Chip label={"" + filterProps.protected} color="primary" onDelete={() => pushParamExact("protected", undefined)} />}
                                     </Stack>
                                 </Card.Item>
                                 <Card.Item>
@@ -260,7 +260,7 @@ export const WorkunitsDashboard: React.FunctionComponent<WorkunitsDashboardProps
                                 <Card.Item>
                                     <Stack horizontal horizontalAlign="space-between">
                                         <Text variant="large" nowrap block styles={{ root: { fontWeight: "bold" } }}>{nlsHPCC.Owner}</Text>
-                                        {filterProps.owner !== undefined && <Chip label={filterProps.owner} clickable color="primary" onDelete={() => pushParamExact("owner", undefined)} />}
+                                        {filterProps.owner !== undefined && <Chip label={filterProps.owner} color="primary" onDelete={() => pushParamExact("owner", undefined)} />}
                                     </Stack>
                                 </Card.Item>
                                 <Card.Item>
@@ -273,7 +273,7 @@ export const WorkunitsDashboard: React.FunctionComponent<WorkunitsDashboardProps
                                 <Card.Item>
                                     <Stack horizontal horizontalAlign="space-between">
                                         <Text variant="large" nowrap block styles={{ root: { fontWeight: "bold" } }}>{nlsHPCC.Cluster}</Text>
-                                        {filterProps.cluster !== undefined && <Chip label={filterProps.cluster} clickable color="primary" onDelete={() => pushParamExact("cluster", undefined)} />}
+                                        {filterProps.cluster !== undefined && <Chip label={filterProps.cluster} color="primary" onDelete={() => pushParamExact("cluster", undefined)} />}
                                     </Stack>
                                 </Card.Item>
                                 <Card.Item>

--- a/esp/src/src-react/components/controls/Chip.tsx
+++ b/esp/src/src-react/components/controls/Chip.tsx
@@ -1,0 +1,127 @@
+import * as React from "react";
+import { styled } from "@fluentui/utilities";
+import { IStyleFunctionOrObject, classNamesFunction } from "@fluentui/react";
+import { IStyle, ITheme } from "@fluentui/style-utilities";
+import { StatusErrorFullIcon } from "@fluentui/react-icons-mdl2";
+
+interface IChipStyleProps {
+    className?: string;
+    theme: ITheme;
+    color: string,
+}
+
+interface IChipStyles {
+    root: IStyle;
+    chipLabel: IStyle;
+    chipCloseIcon: IStyle;
+}
+
+interface IChipProps extends React.RefAttributes<HTMLDivElement> {
+    label: string;
+    color?: string;
+    onDelete?: () => void;
+    className?: string;
+    theme?: ITheme;
+    styles?: IStyleFunctionOrObject<IChipStyleProps, IChipStyles>;
+}
+
+function getStyles(props: IChipStyleProps): IChipStyles {
+
+    const { theme, color, className } = props;
+
+    const classNames = {
+        root: "chip-container",
+        label: "chip-label",
+        closeIcon: "chip-close-icon"
+    };
+
+    let backgroundColor = theme.palette.themeDarker;
+    let textColor = theme.palette.white;
+    let iconColor = theme.palette.themePrimary;
+
+    switch (color) {
+        case "default":
+        case "neutral":
+            backgroundColor = theme.palette.neutralLight;
+            textColor = theme.palette.neutralDark;
+            iconColor = theme.palette.neutralPrimary;
+            break;
+        case "secondary":
+            backgroundColor = theme.palette.themeSecondary;
+            break;
+    }
+
+    const mergedStyles = {
+        root: [
+            classNames.root,
+            {
+                color: textColor,
+                backgroundColor: backgroundColor,
+                height: "30px",
+                minWidth: "5rem",
+                justifyContent: "center",
+                borderRadius: "1.25rem",
+                display: "inline-flex",
+                alignItems: "center"
+            },
+            className
+        ],
+        chipLabel: [
+            classNames.label,
+            {
+                fontSize: "0.8125rem",
+                padding: "0 0.85rem",
+                userSelect: "none"
+            }
+        ],
+        chipCloseIcon: [
+            classNames.closeIcon,
+            {
+                color: iconColor,
+                alignSelf: "baseline",
+                margin: "0.05rem 0.66rem 0 -0.2rem"
+            }
+        ]
+    };
+
+    return mergedStyles;
+}
+
+const getClassNames = classNamesFunction<IChipStyleProps, IChipStyles>();
+
+const ChipBase: React.FunctionComponent<IChipProps> = React.forwardRef<
+    HTMLDivElement,
+    IChipProps
+>((props, forwardedRef) => {
+    
+    const { label, onDelete, theme, color, className, styles } = props;
+
+    const ariaLabel = "Click to remove";
+
+    const classNames = getClassNames(styles!, {
+        theme: theme!,
+        className,
+        color
+    });
+
+    function handleCloseIconClick(evt) {
+        evt.preventDefault();
+        if (onDelete) {
+            onDelete();
+        }
+    }
+
+    return <div className={classNames.root}>
+        <span className={classNames.chipLabel}>{label}</span>
+        { onDelete !== undefined && 
+        <a href="" onClick={handleCloseIconClick} title={ariaLabel} aria-label={ariaLabel} className={classNames.chipCloseIcon}>
+            <StatusErrorFullIcon />
+        </a> }
+    </div>;
+});
+
+export const Chip: React.FunctionComponent<IChipProps> = styled<
+    IChipProps,
+    IChipStyleProps,
+    IChipStyles
+>(ChipBase, getStyles, undefined, { scope: "Chip" });

